pkg/astnormalization/inject_input_default_values_test.go

```diff
@@ -0,0 +1,90 @@
+package astnormalization
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+	"testing"
+)
+
+const testInputDefaultSchema = `
+enum TestEnum {
+  ValueOne
+  ValueTwo
+}
+
+schema {
+  mutation: Mutation
+}
+
+type Mutation {
+  testDefaultValueSimple(data: SimpleTestInput!): String!
+  testNestedInputField(data: InputWithNestedField!): String!
+}
+
+input SimpleTestInput {
+  firstField: String!
+  secondField: Int! = 1
+}
+
+input InputWithNestedField {
+  nested: LowerLevelInput!
+}
+
+input LowerLevelInput {
+  firstField: Int!
+  secondField: TestEnum! = ValueOne
+}
+`
+
+func TestInputDefaultValueExtraction(t *testing.T) {
```

Comment:
Try to add a lot more tests cases covering edge cases with partial default values